Readme.md
# Sixteen

**Contributors:** (inkhive)  
**Requires at least:** WordPress 4.4  
**Tested up to:** WordPress 4.9
**Version:** 1.7.2
**License:** GPL v3  
**License URI:** http://www.gnu.org/licenses/gpl-3.0.html  
**Tags:** custom-background, two-columns, one-column, custom-colors, flexible-header, theme-options, right-sidebar, grid-layout, custom-menu, threaded-comments, translation-ready, featured-images, blog, full-width-template, e-commerce, footer-widgets, entertainment, photography, featured-image-header

Creative and Beautiful WordPress Theme with a Large Header

## Description

Sixteen is a unique, ultra creative multipurpose WordPress theme with complete support for WooCommerce. Sixteen supports all basic features and is fully mobile friendly. And with Support for Custom Widgets, Slider and Featured Posts section, etc you can make this theme look even more beautiful on your website. 
Demo Here: http://demo.inkhive.com/sixteen-plus/


## Copyright


This theme is 100% GPL. And any external resources used and bundled with the theme are also Fully Compatible with the GPL.

1. Slicknav
	- MIT License
	- https://github.com/ComputerWolf/SlickNav

2. _S/Underscores Framework
	- GPL v2
	- http://underscores.me

3. nivoSlider
    - MIT License
    - http://nivo.dev7studios.com

4. Retina.js
    - MIT License
    - https://github.com/imulus/retinajs
    
All Other Resources, apart from the ones mentioned above have been created by me fall under GPL v3 license of the theme.	

## Screenshot & Image Credits

* http://pixabay.com/static/uploads/photo/2014/07/25/08/48/sunset-401541_150.jpg
* http://pixabay.com/static/uploads/photo/2014/07/10/10/20/golden-gate-bridge-388917_150.jpg
* http://pixabay.com/static/uploads/photo/2014/08/11/11/05/oriental-pearl-tower-415474_150.jpg
* http://pixabay.com/static/uploads/photo/2014/07/01/12/37/lifeguard-381240_150.jpg
* http://pixabay.com/static/uploads/photo/2014/07/01/12/35/taxi-cab-381233_150.jpg
* http://pixabay.com/static/uploads/photo/2014/07/01/12/36/smartphone-381237_150.jpg
* http://pixabay.com/en/photographer-tourist-snapshot-407068/
* http://pixabay.com/en/milsons-point-sydney-australia-330400/
* http://unsplash.com/post/94204091269/download-by-elaine-li
* http://pixabay.com/en/manhattan-new-york-city-407703/

License: CC0
Source: https://pixabay.com/en/service/terms/	
        http://creativecommons.org/publicdomain/zero/1.0/

## Changelog
	
### Version 1.0

* Intial Release
	
### Version 1.0.1

* Transparent #content backgroung
* Fixed Long Words overflowing container problem.
* Updated Screenshot	
	
### Version 1.0.2 

* Updated Screenshot

### Version 1.0.3 

* Fixed text-domain problems.
* Escaped Social Links	

### Version 1.0.5

* Fixed and Issue which caused theme to be initialized with white background.
* Minor Bug fixes.
* Fixed Issues with Static Front Pages. Now sidebar shows at desingated position.
	
### Version 1.0.6

* Fixed and Issue which caused the main.css file to not load in the demo, thus breaking the entire site.	
	
### Version 1.0.7

* Fixed and issue which arised due to previous updates, resulting in homepage posts not being shown.	
	
### Version 1.0.8

* Fixed Translation Issues.
* Added Support for French and Spanish.	
	
### Version 1.0.9
	
* Corrected the contents of sixteen.pot and sixteen.po	
	
### Version 1.1

* This is a major update. 
* Footer Siderbar added.
* Social Icons added.
* Responsive Designed enhanced.
* Minor bug fixes.
* Associated with pro version.	
	
### Version 1.1.1

* Fixed bug with Nivo Slider.
* Added link to Demo Blog.
* Other Minor Bug Fixes.
	
### Version 1.1.2 & 1.1.2.1

* Fixed Bugs
* Edited Options Panel
* Provided Solution to Social Network Icon issue
* Fixed Style Issue
	
### Version 1.1.3

* Using a Fully GPL Screenshot
* Social Links open in new Window
* Updated Pro Version Details	
		
### Version 1.1.3.1

* Updated Discount Details
* Added User translated .po files for Arabic, Russian & Japanese.	
	
### Version 1.2

* Updated Theme Tags(Added all tags, this theme supports)
* Updated Coupon Terms, added Expiry date.
* Default Image Issues Fixed. 
* Switched a Different Default Image.	
	
### Version 1.2.0.1

* Fixed overflow Issue with Full Width Pages	
* Removed Search Widget from Default Sidebar
* Reduced the Superfish Menu Disappearance time.
* Removed Expired Coupon Details	
	
### Version 1.2.0.2

* Fixed a Major Issue, which was causing Sidebar to not load in Child Themes.	
	
### Version 1.2.1 

* Major Security Updated.
* Theme was found to be vulnerable to Dangerous Javascript Insertion via Theme Options.
* Custom Header/Footer Scripts Removed.
* Textarea Field Santization Done.
* All Output URL properly escaped.
* Tested against all the recent guidelines.	
		   
### Version 1.2.1.3

* Extended support for Child Themes, Removed Filter on Header		
	
### Version 1.2.1.4

* Minor Changes to CSS Code
* Improved Behaviour when Header Text is disabled.	 
	
### Version 1.2.1.5

* Changed the way Slider is rendered	  
	
### Version 1.2.1.6

* Relocated Code for Social Icons.
	
### Version 1.2.1.7

* Added Developer Reference for Nivo Slider
* Added Numbered Page Navigation for Search Results	
	
### Version 1.2.1.8

* Removed image.php to Make Image Attachments render using default template.	
		
### Version 1.2.1.9

* Optimized Default Images for Faster Loading of Default Content and Demo Blogs.
	
### Version 1.2.2

* Fixed $content_width Issues.
* Fixed Issues with Font on Some Browsers.
	
### Version 1.3 (Major Update)

* WooCommerce Integrated.
* Slider Rebuilt with Better Animations.
* Social Icons ReDone.
* Fixed Display Issues on iPhone 5/5s.	
* Other Minor Fixes.
* Fixed Issue with Header Image.	
* Added defaults.		
	
### Version 1.3.0.2

* Bug Fix for WooCommerce Cart Page, Cross Sells Item do not Collapse anymore.
* Style for WooCommerce Buttons Added.
	
### Version 1.3.0.3

* Removed Codekit.config
* Fixed Slider Pagination Bug. Background Appeared different on Some Screens.
* Added New Page Template, for Page without Titles.	
	
### Version 1.3.0.4

* Corre&&&Version 1.5
* Updated to Customizer. Removed Options Framework.
### Updated WooCommerce Files.	
	
### Version 1.5.1

* Fixed bugs Related to Slider.
* Corrected a Few URLS.	
	
### Version 1.5.2

* Bug Fixes Related to Links.
* Removed Ref. to seller theme.	
* Logo Issues Fixed.
	
### Version 1.5.3

* Added a New Page Template	
	
### Version 1.5.4

* Slider now works on Front Page too.	
	
### Version 1.5.5

* Minor Bug Fixes.
* Title tag now supported	

### Version 1.5.6
	
* WooCommerce Update Fix
* Added Support for WooCommerce Product Gallery Lightbox Plugin

### Version 16.3

* Switched to new Framework

### Version 1.6.4

* Category Posts Issue Fixed
* Archive Page Posts Issue Fixed

### Version: 1.7.0

* Custom logo added.
* Admin panel minimized.
* Header Image bug fixed.
* Site title and tagline color bug fixed.
* Featured post boxes area major bug fixed.
* Featured post large area major bug fixed.
* Default blog layout bug fixed.
* Slider nivo major bug fixed.
* Main navigation and Mobile navigation major bug fixed.
* Other minor bug fixed.

### Version: 1.7.1

* Search form unwanted issue fixed.

### Version 1.7.2

* Extra Custom CSS Control Removed
* Documentation & Theme Support Added
* Minor Bug Fixes



changelog.txt
Version 1.0

	i) Intial Release
	
Version 1.0.1

	i) Transparent #content backgroung
	ii) Fixed Long Words overflowing container problem.
	iii) Updated Screenshot	
	
Version 1.0.2 

	i) Updated Screenshot

Version 1.0.3 

	i) Fixed text-domain problems.
	ii) Escaped Social Links	

Version 1.0.5

	i) Fixed and Issue which caused theme to be initialized with white background.
	ii) Minor Bug fixes.
	iii) Fixed Issues with Static Front Pages. Now sidebar shows at desingated position.
	
Version 1.0.6

	i) Fixed and Issue which caused the main.css file to not load in the demo, thus breaking the entire site.	
	
Version 1.0.7

	i) Fixed and issue which arised due to previous updates, resulting in homepage posts not being shown.	
	
Version 1.0.8

	i) Fixed Translation Issues.
	ii) Added Support for French and Spanish.	
	
Version 1.0.9
	
	i) Corrected the contents of sixteen.pot and sixteen.po	
	
Version 1.1

	i) This is a major update. 
	ii) Footer Siderbar added.
	iii) Social Icons added.
	iv) Responsive Designed enhanced.
	v) Minor bug fixes.
	vi) Associated with pro version.	
	
Version 1.1.1

	i) Fixed bug with Nivo Slider.
	ii) Added link to Demo Blog.
	iii) Other Minor Bug Fixes.
	
Version 1.1.2 & 1.1.2.1

	i) Fixed Bugs
	ii) Edited Options Panel
	iii) Provided Solution to Social Network Icon issue
	iv) Fixed Style Issue
	
Version 1.1.3

	i) Using a Fully GPL Screenshot
	ii) Social Links open in new Window
	iii) Updated Pro Version Details	
		
Version 1.1.3.1

	i) Updated Discount Details
	ii) Added User translated .po files for Arabic, Russian & Japanese.	
	
Version 1.2

	i) Updated Theme Tags(Added all tags, this theme supports)
	ii)	Updated Coupon Terms, added Expiry date.
	iii) Default Image Issues Fixed. 
	iv) Switched a Different Default Image.	
	
Version 1.2.0.1

	i) Fixed overflow Issue with Full Width Pages	
	ii) Removed Search Widget from Default Sidebar
	iii) Reduced the Superfish Menu Disappearance time.
	iv) Removed Expired Coupon Details	
	
Version 1.2.0.2

	i) Fixed a Major Issue, which was causing Sidebar to not load in Child Themes.	
	
Version 1.2.1 

	i) Major Security Updated.
		- Theme was found to be vulnerable to Dangerous Javascript Insertion via Theme Options.
		To Avoid that issue,
		   - Custom Header/Footer Scripts Removed.
		   - Textarea Field Santization Done.
		   - All Output URL properly escaped.
		   - Tested against all the recent guidelines.	
		   
Version 1.2.1.3

	i) Extended support for Child Themes, Removed Filter on Header		
	
Version 1.2.1.4

	i) Minor Changes to CSS Code
	ii) Improved Behaviour when Header Text is disabled.	 
	
Version 1.2.1.5

	i) Changed the way Slider is rendered	  
	
Version 1.2.1.6

	i) Relocated Code for Social Icons.
	
Version 1.2.1.7

	i) Added Developer Reference for Nivo Slider
	ii) Added Numbered Page Navigation for Search Results	
	
Version 1.2.1.8

	i) Removed image.php to Make Image Attachments render using default template.	
		
Version 1.2.1.9

	i) Optimized Default Images for Faster Loading of Default Content and Demo Blogs.
	
Version 1.2.2

	i) Fixed $content_width Issues.
	ii) Fixed Issues with Font on Some Browsers.
	
Version 1.3 (Major Update)

	i) WooCommerce Integrated.
	ii) Slider Rebuilt with Better Animations.
	iii) Social Icons ReDone.
	iv) Fixed Display Issues on iPhone 5/5s.	
	v) Other Minor Fixes.
	vi) Fixed Issue with Header Image.	
	vii) Added defaults.		
	
Version 1.3.0.2

	i) Bug Fix for WooCommerce Cart Page, Cross Sells Item do not Collapse anymore.
	ii) Style for WooCommerce Buttons Added.
	
Version 1.3.0.3

	i) Removed Codekit.config
	ii) Fixed Slider Pagination Bug. Background Appeared different on Some Screens.
	iii) Added New Page Template, for Page without Titles.	
	
Version 1.3.0.4

	i) Corrected Demo Blog URL
	ii) Removed Unused Hook in Header.
	iii) Other Minor Code Modifications. No Functionality Change.	
	iv) Removed screen_icon();	
	
Version 1.3.0.5

	i) Removed Instagram unused icon.
	ii) Compressed Theme Social Icons, total size reduced by 30+ KB.

Version 1.3.0.6

	i) Corrected Issues reported with Language file.
	ii) Renamed changelog file.	
	iii) Removed Unwanted Description Title for Product Pages.
	
Version 1.5

	i) Updated to Customizer. Removed Options Framework.
	ii) Updated WooCommerce Files.	
	
Version 1.5.1

	i) Fixed bugs Related to Slider.
	ii) Corrected a Few URLS.	
	
Version 1.5.2

	i) Bug Fixes Related to Links.
	ii) Removed Ref. to seller theme.	
	iii) Logo Issues Fixed.
	
Version 1.5.3

	i) Added a New Page Template	
	
Version 1.5.4

	i) Slider now works on Front Page too.	
	
Version 1.5.5

	i) Minor Bug Fixes.
	ii) Title tag now supported	

Version 1.5.6
	
	i) WooCommerce Update Fix
	ii) Added Support for WooCommerce Product Gallery Lightbox Plugin

Version 16.3

	i) Switched to new Framework

Version 1.6.4

    i) Category Posts Issue Fixed
    ii) Archive Page Posts Issue Fixed

Version: 1.7.0

    i) Custom logo added.
    ii) Admin panel minimized.
    iii) Header Image bug fixed.
    iv) Site title and tagline color bug fixed.
    v) Featured post boxes area major bug fixed.
    vi) Featured post large area major bug fixed.
    vii) Default blog layout bug fixed.
    viii) Slider nivo major bug fixed.
    ix) Main navigation and Mobile navigation major bug fixed.
    x) Other minor bug fixed.

Version: 1.7.1

    i) Search form unwanted issue fixed.


inc/_customizer.php
<?php
/**
 * sixteen Theme Customizer
 *
 * @package sixteen
 */

/**
 * Add postMessage support for site title and description for the Theme Customizer.
 *
 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
 */
function sixteen_customize_register( $wp_customize ) {
    $wp_customize->get_setting( 'blogname' )->transport         = 'postMessage';
    $wp_customize->get_setting( 'blogdescription' )->transport  = 'postMessage';



    //Logo Settings
    $wp_customize->add_section( 'title_tagline' , array(
        'title'      => __( 'Title, Tagline & Logo', 'sixteen' ),
        'priority'   => 30,
    ) );

    $wp_customize->add_setting( 'sixteen_logo' , array(
        'default'     => '',
        'sanitize_callback' => 'esc_url_raw',
    ) );
    $wp_customize->add_control(
        new WP_Customize_Image_Control(
            $wp_customize,
            'sixteen_logo',
            array(
                'label' => __('Upload Logo','sixteen'),
                'section' => 'title_tagline',
                'settings' => 'sixteen_logo',
                'priority' => 5,
            )
        )
    );


    //Replace Header Text Color with, separate colors for Title and Description
    //Override sixteen_site_titlecolor
    $wp_customize->remove_control('display_header_text');
    $wp_customize->remove_setting('header_textcolor');
    $wp_customize->add_setting('sixteen_site_titlecolor', array(
        'default'     => '#3a85ae',
        'sanitize_callback' => 'sanitize_hex_color',
    ));

    $wp_customize->add_control(new WP_Customize_Color_Control(
            $wp_customize,
            'sixteen_site_titlecolor', array(
            'label' => __('Site Title Color','sixteen'),
            'section' => 'colors',
            'settings' => 'sixteen_site_titlecolor',
            'type' => 'color'
        ) )
    );

    //Settings For Logo Area

    function sixteen_title_visible( $control ) {
        $option = $control->manager->get_setting('sixteen_hide_title_tagline');
        return $option->value() == false ;
    }

    // SLIDER PANEL
    $wp_customize->add_panel( 'sixteen_slider_panel', array(
        'priority'       => 35,
        'capability'     => 'edit_theme_options',
        'theme_supports' => '',
        'title'          => 'Main Slider',
    ) );

    $wp_customize->add_section(
        'sixteen_sec_slider_options',
        array(
            'title'     => __('Enable/Disable','sixteen'),
            'priority'  => 0,
            'panel'     => 'sixteen_slider_panel'
        )
    );


    $wp_customize->add_setting(
        'sixteen_main_slider_enable',
        array( 'sanitize_callback' => 'sixteen_sanitize_checkbox' )
    );

    $wp_customize->add_control(
        'sixteen_main_slider_enable', array(
            'settings' => 'sixteen_main_slider_enable',
            'label'    => __( 'Enable Slider.', 'sixteen' ),
            'section'  => 'sixteen_sec_slider_options',
            'type'     => 'checkbox',
        )
    );

    $wp_customize->add_setting(
        'sixteen_main_slider_count',
        array(
            'default' => '0',
            'sanitize_callback' => 'sixteen_sanitize_positive_number'
        )
    );

    // Select How Many Slides the User wants, and Reload the Page.
    $wp_customize->add_control(
        'sixteen_main_slider_count', array(
            'settings' => 'sixteen_main_slider_count',
            'label'    => __( 'No. of Slides(Min:0, Max: 5)' ,'sixteen'),
            'section'  => 'sixteen_sec_slider_options',
            'type'     => 'number',
            'description' => __('Save the Settings, and Reload this page to Configure the Slides.','sixteen'),

        )
    );



    if ( get_theme_mod('sixteen_main_slider_count') > 0 ) :
        $slides = get_theme_mod('sixteen_main_slider_count');

        for ( $i = 1 ; $i <= $slides ; $i++ ) :

            //Create the settings Once, and Loop through it.

            $wp_customize->add_setting(
                'sixteen_slide_img'.$i,
                array( 'sanitize_callback' => 'esc_url_raw' )
            );

            $wp_customize->add_control(
                new WP_Customize_Image_Control(
                    $wp_customize,
                    'sixteen_slide_img'.$i,
                    array(
                        'label' => '',
                        'section' => 'sixteen_slide_sec'.$i,
                        'settings' => 'sixteen_slide_img'.$i,
                    )
                )
            );


            $wp_customize->add_section(
                'sixteen_slide_sec'.$i,
                array(
                    'title'     => 'Slide '.$i,
                    'priority'  => $i,
                    'panel'     => 'sixteen_slider_panel'
                )
            );

            $wp_customize->add_setting(
                'sixteen_slide_title'.$i,
                array( 'sanitize_callback' => 'sanitize_text_field' )
            );

            $wp_customize->add_control(
                'sixteen_slide_title'.$i, array(
                    'settings' => 'sixteen_slide_title'.$i,
                    'label'    => __( 'Slide Title','sixteen' ),
                    'section'  => 'sixteen_slide_sec'.$i,
                    'type'     => 'text',
                )
            );

            $wp_customize->add_setting(
                'sixteen_slide_desc'.$i,
                array( 'sanitize_callback' => 'sanitize_text_field' )
            );

            $wp_customize->add_control(
                'sixteen_slide_desc'.$i, array(
                    'settings' => 'sixteen_slide_desc'.$i,
                    'label'    => __( 'Slide Description','sixteen' ),
                    'section'  => 'sixteen_slide_sec'.$i,
                    'type'     => 'text',
                )
            );


            $wp_customize->add_setting(
                'sixteen_slide_url'.$i,
                array( 'sanitize_callback' => 'esc_url_raw' )
            );

            $wp_customize->add_control(
                'sixteen_slide_url'.$i, array(
                    'settings' => 'sixteen_slide_url'.$i,
                    'label'    => __( 'Target URL','sixteen' ),
                    'section'  => 'sixteen_slide_sec'.$i,
                    'type'     => 'url',
                )
            );

        endfor;


    endif;


    //Sidebar Location
    $wp_customize->add_section(
        'sixteen_sidebar_options',
        array(
            'title'     => __('Sidebar Locations','sixteen'),
            'priority'  => 41,
        )
    );


    $wp_customize->add_setting(
        'sixteen_left_sidebar',
        array( 'sanitize_callback' => 'sixteen_sanitize_checkbox' )
    );

    $wp_customize->add_control(
        'sixteen_left_sidebar', array(
            'settings' => 'sixteen_left_sidebar',
            'label'    => __( 'Switch Sidebar to Left Side', 'sixteen' ),
            'section'  => 'sixteen_sidebar_options',
            'type'     => 'checkbox',
        )
    );



    if (class_exists('WP_Customize_Control')) {
        class sixteen_WP_Customize_Upgrade_Control extends WP_Customize_Control {
            /**
             * Render the control's content.
             */
            public function render_content() {
                printf(
                    '<label class="customize-control-upgrade"><span class="customize-control-title">%s</span> %s</label>',
                    $this->label,
                    $this->description
                );
            }
        }
    }


    //Upgrade
    $wp_customize->add_section(
        'sixteen_sec_upgrade',
        array(
            'title'     => __('Discover SIXTEEN PLUS','sixteen'),
            'priority'  => 45,
        )
    );

    $wp_customize->add_setting(
        'sixteen_upgrade',
        array( 'sanitize_callback' => 'esc_textarea' )
    );

    $wp_customize->add_control(
        new sixteen_WP_Customize_Upgrade_Control(
            $wp_customize,
            'sixteen_upgrade',
            array(
                'label' => __('More of Everything','sixteen'),
                'description' => __('Sixteen Plus has more of Everything. More New Features, More Options, Unlimited Colors, More Fonts, More Layouts, Configurable Slider, Inbuilt Advertising Options, More Widgets, and a lot more options and comes with Dedicated Support. To Know More about the Pro Version, click here: <a href="http://inkhive.com/product/sixteen-plus/">Upgrade to Pro Version</a>.','sixteen'),
                'section' => 'sixteen_sec_upgrade',
                'settings' => 'sixteen_upgrade',
            )
        )
    );


    class sixteen_Custom_CSS_Control extends WP_Customize_Control {
        public $type = 'textarea';

        public function render_content() {
            ?>
            <label>
                <span class="customize-control-title"><?php echo esc_html( $this->label ); ?></span>
                <textarea rows="8" style="width:100%;" <?php $this->link(); ?>><?php echo esc_textarea( $this->value() ); ?></textarea>
            </label>
            <?php
        }
    }

    $wp_customize-> add_section(
        'sixteen_custom_codes',
        array(
            'title'			=> __('Custom CSS','sixteen'),
            'description'	=> __('Enter your Custom CSS to Modify design.','sixteen'),
            'priority'		=> 41,
        )
    );

    $wp_customize->add_setting(
        'sixteen_custom_css',
        array(
            'default'		=> '',
            'capability'           => 'edit_theme_options',
            'sanitize_callback'    => 'wp_filter_nohtml_kses',
            'sanitize_js_callback' => 'wp_filter_nohtml_kses'
        )
    );

    $wp_customize->add_control(
        new sixteen_Custom_CSS_Control(
            $wp_customize,
            'sixteen_custom_css',
            array(
                'section' => 'sixteen_custom_codes',


            )
        )
    );

    function sixteen_sanitize_text( $input ) {
        return wp_kses_post( force_balance_tags( $input ) );
    }

    $wp_customize-> add_section(
        'sixteen_custom_footer',
        array(
            'title'			=> __('Custom Footer Text','sixteen'),
            'description'	=> __('Enter your Own Copyright Text.','sixteen'),
            'priority'		=> 42,
        )
    );

    $wp_customize->add_setting(
        'sixteen_footer_text',
        array(
            'default'		=> '',
            'sanitize_callback'	=> 'sanitize_text_field'
        )
    );

    $wp_customize->add_control(
        'sixteen_footer_text',
        array(
            'section' => 'sixteen_custom_footer',
            'settings' => 'sixteen_footer_text',
            'type' => 'text'
        )
    );


    // Social Icons
    $wp_customize->add_section('sixteen_social_section', array(
        'title' => __('Social Icons','sixteen'),
        'priority' => 44 ,
    ));

    $social_networks = array( //Redefinied in Sanitization Function.
        'none' => __('-','sixteen'),
        'facebook' => __('Facebook','sixteen'),
        'twitter' => __('Twitter','sixteen'),
        'google-plus' => __('Google Plus','sixteen'),
        'instagram' => __('Instagram','sixteen'),
        'rss' => __('RSS Feeds','sixteen'),
        'flickr' => __('Flickr','sixteen'),
    );

    $social_count = count($social_networks);

    for ($x = 1 ; $x <= ($social_count - 3) ; $x++) :

        $wp_customize->add_setting(
            'sixteen_social_'.$x, array(
            'sanitize_callback' => 'sixteen_sanitize_social',
            'default' => 'none'
        ));

        $wp_customize->add_control( 'sixteen_social_'.$x, array(
            'settings' => 'sixteen_social_'.$x,
            'label' => __('Icon ','sixteen').$x,
            'section' => 'sixteen_social_section',
            'type' => 'select',
            'choices' => $social_networks,
        ));

        $wp_customize->add_setting(
            'sixteen_social_url'.$x, array(
            'sanitize_callback' => 'esc_url_raw'
        ));

        $wp_customize->add_control( 'sixteen_social_url'.$x, array(
            'settings' => 'sixteen_social_url'.$x,
            'description' => __('Icon ','sixteen').$x.__(' Url','sixteen'),
            'section' => 'sixteen_social_section',
            'type' => 'url',
            'choices' => $social_networks,
        ));

    endfor;

    function sixteen_sanitize_social( $input ) {
        $social_networks = array(
            'none' ,
            'facebook',
            'twitter',
            'google-plus',
            'instagram',
            'rss',
            'flickr',
        );
        if ( in_array($input, $social_networks) )
            return $input;
        else
            return '';
    }


    /* Sanitization Functions Common to Multiple Settings go Here, Specific Sanitization Functions are defined along with add_setting() */
    function sixteen_sanitize_checkbox( $input ) {
        if ( $input == 1 ) {
            return 1;
        } else {
            return '';
        }
    }

    function sixteen_sanitize_positive_number( $input ) {
        if ( ($input >= 0) && is_numeric($input) )
            return $input;
        else
            return '';
    }

    function sixteen_sanitize_category( $input ) {
        if ( term_exists(get_cat_name( $input ), 'category') )
            return $input;
        else
            return '';
    }


}
add_action( 'customize_register', 'sixteen_customize_register' );


/**
 * Binds JS handlers to make Theme Customizer preview reload changes asynchronously.
 */
function sixteen_customize_preview_js() {
    wp_enqueue_script( 'sixteen_customizer', get_template_directory_uri() . '/js/customizer.js', array( 'customize-preview' ), '20130508', true );
}
add_action( 'customize_preview_init', 'sixteen_customize_preview_js' );
